### PR TITLE
feat: add custom keystore/truststore config for zeebe client

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

feat: add custom keystore/truststore config for zeebe client

There are cases where a user doesn't want to maintain whole SSL infrastructure for the Camunda cluster but also use X509 authorizes.

This code allows configure SSL context only against authorizer HTTP call.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to: https://github.com/camunda/product-hub/issues/2451

### CODE REVIEW

**Comment by igpetrov** [comment]
Hi @npepinpe 

This one is ready for review

**Comment by npepinpe** [comment]
@nicpuppa - can you take care of the review please? :pray: I don't think we need more than 1 review, and I trust you can do it well ;)

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java [hunk 1]
```diff
@@ -170,4 +191,41 @@ private ZeebeClientCredentials fetchCredentials() throws IOException {
...
   }
+
+  private void maybeConfigureCustomSSLContext(final HttpURLConnection connection) {
+    if (connection instanceof HttpsURLConnection) {
+      final HttpsURLConnection httpsConnection = (HttpsURLConnection) connection;
+      httpsConnection.setSSLSocketFactory(createSSLContext());
+    }
+  }
+
+  private SSLSocketFactory createSSLContext() {
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:202]: 🔧 Just a little refactor

```suggestion
  private SSLSocketFactory createSSLContext() {
    if (keystorePath == null && truststorePath == null) {
      return SSLContexts.createSystemDefault().getSocketFactory();
    }
    final SSLContextBuilder builder = SSLContexts.custom();
    try {
      if (keystorePath != null) {
        builder.loadKeyMaterial(
            keystorePath,
            keystorePassword == null ? null : keystorePassword.toCharArray(),
            keystoreKeyPassword == null ? new char[0] : keystoreKeyPassword.toCharArray());
      }

      if (truststorePath != null) {
        builder.loadTrustMaterial(
            truststorePath, truststorePassword == null ? null : truststorePassword.toCharArray());
      }

      return builder.build().getSocketFactory();

    } catch (final NoSuchAlgorithmException 
                   | KeyManagementException 
                   | KeyStoreException 
                   | UnrecoverableKeyException 
                   | CertificateException 
                   | IOException e) {
      throw new RuntimeException("Failed to create SSL context", e);
    }
  }
```

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProviderBuilder.java [hunk 2]
```diff
@@ -203,6 +291,26 @@ private void checkEnvironmentOverrides() {
...
+    if (envKeystorePath != null) {
+      keystorePath = Paths.get(envKeystorePath);
+    }
+
+    if (envKeystorePassword != null) {
+      keystorePassword = envKeystorePassword;
+    }
+
+    if (envKeystoreKeyPassword != null) {
+      keystoreKeyPassword = envKeystorePassword;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProviderBuilder.java:303]: ❌ 

```suggestion
      keystoreKeyPassword = envKeystoreKeyPassword;
```

**Review by nicpuppa — CHANGES_REQUESTED** [review]
Thanks @igpetrov , looks good. Please have a look at my comments.

❓ Do we want to support this new configuration also in the Spring SDK

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProviderBuilder.java (hunk 2 above)
Comment (igpetrov) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProviderBuilder.java]: Ahh, good catch! Fixed

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java (hunk 1 above)
Comment (igpetrov) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:202]: Applied your suggestion. Thank you!

**Comment by igpetrov** [comment]
@nicpuppa thank you for finding time for this review.

> ❓ Do we want to support this new configuration also in the Spring SDK

Probably, yes, but if so, I will do it in a separate PR.

**Review by nicpuppa — APPROVED** [review]

**Pull request merged**